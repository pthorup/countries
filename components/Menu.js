import { useState, useEffect } from 'react'
import Link from 'next/link'
import Image from 'next/image'
import { ApolloClient, InMemoryCache, gql } from '@apollo/client'
import styles from '../styles/Menu.module.css'

const Menu = () => {
   const [showSubMenu, setShowSubMenu] = useState('none')
   const [continents, setContinents] = useState([])

   useEffect(async () => {
      const client = new ApolloClient({
         uri: 'https://countries.trevorblades.com/',
         cache: new InMemoryCache(),
      })
      const { data } = await client.query({
         query: gql`
            {
               continents {
                  name
                  code
               }
            }
         `,
      })
      setContinents(data.continents)
   }, [])

   const handleMenuToggle = () => {
      setShowSubMenu((prev) => (prev === 'none' ? 'block' : 'none'))
   }

   return (
      <div className={styles.main}>
         <div className={styles.logo}>
            <Image
               src='/logo.png'
               alt='Fun Learning Countries'
               width={600}
               height={419}
            />
         </div>

         <ul className={styles.menu}>
            <li className={styles.menuLink}>
               <Link href='/'>
                  <a>Home</a>
               </Link>
            </li>
            <li className={styles.menuLink}>
               <span className={styles.subMenuTitle} onClick={handleMenuToggle}>
                  Continents
               </span>

               <ul
                  className={styles.subMenu}
                  style={{ display: `${showSubMenu}` }}
               >
                  {continents.map((continent) => (
                     <li key={continent.code} className={styles.menuLink}>
                        <Link href={`/continent/${continent.code}`}>
                           <a>{continent.name}</a>
                        </Link>
                     </li>
                  ))}
               </ul>
            </li>
            <li className={styles.menuLink}>
               <Link href='/guess-country'>
                  <a>Play A Game</a>
               </Link>
            </li>
         </ul>
      </div>
   )
}

export default Menu
