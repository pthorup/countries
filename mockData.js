const answer = 'Sweden'

const continent = {
   name: 'Africa',
   countries: [
      {
         __typename: 'Country',
         name: 'Angola',
         code: 'AO',
      },
      {
         __typename: 'Country',
         name: 'Burkina Faso',
         code: 'BF',
      },
      {
         __typename: 'Country',
         name: 'Burundi',
         code: 'BI',
      },
      {
         __typename: 'Country',
         name: 'Benin',
         code: 'BJ',
      },
      {
         __typename: 'Country',
         name: 'Botswana',
         code: 'BW',
      },
      {
         __typename: 'Country',
         name: 'Democratic Republic of the Congo',
         code: 'CD',
      },
      {
         __typename: 'Country',
         name: 'Central African Republic',
         code: 'CF',
      },
      {
         __typename: 'Country',
         name: 'Republic of the Congo',
         code: 'CG',
      },
      {
         __typename: 'Country',
         name: 'Ivory Coast',
         code: 'CI',
      },
      {
         __typename: 'Country',
         name: 'Cameroon',
         code: 'CM',
      },
      {
         __typename: 'Country',
         name: 'Cape Verde',
         code: 'CV',
      },
      {
         __typename: 'Country',
         name: 'Djibouti',
         code: 'DJ',
      },
      {
         __typename: 'Country',
         name: 'Algeria',
         code: 'DZ',
      },
      {
         __typename: 'Country',
         name: 'Egypt',
         code: 'EG',
      },
      {
         __typename: 'Country',
         name: 'Western Sahara',
         code: 'EH',
      },
      {
         __typename: 'Country',
         name: 'Eritrea',
         code: 'ER',
      },
      {
         __typename: 'Country',
         name: 'Ethiopia',
         code: 'ET',
      },
      {
         __typename: 'Country',
         name: 'Gabon',
         code: 'GA',
      },
      {
         __typename: 'Country',
         name: 'Ghana',
         code: 'GH',
      },
      {
         __typename: 'Country',
         name: 'Gambia',
         code: 'GM',
      },
      {
         __typename: 'Country',
         name: 'Guinea',
         code: 'GN',
      },
      {
         __typename: 'Country',
         name: 'Equatorial Guinea',
         code: 'GQ',
      },
      {
         __typename: 'Country',
         name: 'Guinea-Bissau',
         code: 'GW',
      },
      {
         __typename: 'Country',
         name: 'Kenya',
         code: 'KE',
      },
      {
         __typename: 'Country',
         name: 'Comoros',
         code: 'KM',
      },
      {
         __typename: 'Country',
         name: 'Liberia',
         code: 'LR',
      },
      {
         __typename: 'Country',
         name: 'Lesotho',
         code: 'LS',
      },
      {
         __typename: 'Country',
         name: 'Libya',
         code: 'LY',
      },
      {
         __typename: 'Country',
         name: 'Morocco',
         code: 'MA',
      },
      {
         __typename: 'Country',
         name: 'Madagascar',
         code: 'MG',
      },
      {
         __typename: 'Country',
         name: 'Mali',
         code: 'ML',
      },
      {
         __typename: 'Country',
         name: 'Mauritania',
         code: 'MR',
      },
      {
         __typename: 'Country',
         name: 'Mauritius',
         code: 'MU',
      },
      {
         __typename: 'Country',
         name: 'Malawi',
         code: 'MW',
      },
      {
         __typename: 'Country',
         name: 'Mozambique',
         code: 'MZ',
      },
      {
         __typename: 'Country',
         name: 'Namibia',
         code: 'NA',
      },
      {
         __typename: 'Country',
         name: 'Niger',
         code: 'NE',
      },
      {
         __typename: 'Country',
         name: 'Nigeria',
         code: 'NG',
      },
      {
         __typename: 'Country',
         name: 'Réunion',
         code: 'RE',
      },
      {
         __typename: 'Country',
         name: 'Rwanda',
         code: 'RW',
      },
      {
         __typename: 'Country',
         name: 'Seychelles',
         code: 'SC',
      },
      {
         __typename: 'Country',
         name: 'Sudan',
         code: 'SD',
      },
      {
         __typename: 'Country',
         name: 'Saint Helena',
         code: 'SH',
      },
      {
         __typename: 'Country',
         name: 'Sierra Leone',
         code: 'SL',
      },
      {
         __typename: 'Country',
         name: 'Senegal',
         code: 'SN',
      },
      {
         __typename: 'Country',
         name: 'Somalia',
         code: 'SO',
      },
      {
         __typename: 'Country',
         name: 'South Sudan',
         code: 'SS',
      },
      {
         __typename: 'Country',
         name: 'São Tomé and Príncipe',
         code: 'ST',
      },
      {
         __typename: 'Country',
         name: 'Swaziland',
         code: 'SZ',
      },
      {
         __typename: 'Country',
         name: 'Chad',
         code: 'TD',
      },
      {
         __typename: 'Country',
         name: 'Togo',
         code: 'TG',
      },
      {
         __typename: 'Country',
         name: 'Tunisia',
         code: 'TN',
      },
      {
         __typename: 'Country',
         name: 'Tanzania',
         code: 'TZ',
      },
      {
         __typename: 'Country',
         name: 'Uganda',
         code: 'UG',
      },
      {
         __typename: 'Country',
         name: 'Mayotte',
         code: 'YT',
      },
      {
         __typename: 'Country',
         name: 'South Africa',
         code: 'ZA',
      },
      {
         __typename: 'Country',
         name: 'Zambia',
         code: 'ZM',
      },
      {
         __typename: 'Country',
         name: 'Zimbabwe',
         code: 'ZW',
      },
   ],
}

module.exports = {
   answer,
   continent,
}
