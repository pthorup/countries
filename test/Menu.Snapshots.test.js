import React from 'react'
import Menu from '../components/Menu'
import renderer from 'react-test-renderer'
import { answer } from '../mockData'

it('Menu links', () => {
   const tree = renderer.create(<Menu />)

   expect(tree).toMatchSnapshot()
})
