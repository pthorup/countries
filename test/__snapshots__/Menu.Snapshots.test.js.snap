// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu links 1`] = `
<ul
  className="menu"
>
  <li
    className="menuLink"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="menuLink"
  >
    <a
      className="subMenuTitle"
      onClick={[Function]}
    >
      Continents
    </a>
    <ul
      className="subMenu"
      style={
        Object {
          "display": "none",
        }
      }
    />
  </li>
  <li
    className="menuLink"
  >
    <a
      href="/guess-country"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Play A Game
    </a>
  </li>
</ul>
`;
